spec/presenters/submission_presenter_spec.rb

```diff
@@ -1,68 +1,89 @@
 require 'rails_helper'
 
 RSpec.describe SubmissionPresenter do
-  subject { described_class.new(submission, rates, submissions_repository) }
-  let!(:rates) { submission.rates }
-  let!(:submissions_repository) { double }
+  let(:submission) { instance_double(Submission) }
+  let(:submission_repository) { double }
+  let(:submission_presenter) { described_class.new(submission, rates, submission_repository) }
 
-  describe do
-    let!(:submission) { FactoryGirl.build(:submission) }
+  describe "is a delegator" do
+    let(:rates) { double }
+    subject { submission_presenter }
 
     it "acts like a Submission" do
-      expect(subject).to eq(submission)
-      expect(subject).not_to be_a(Submission)
+      is_expected.to eq(submission)
+      is_expected.not_to be_a(Submission)
     end
   end
 
   describe "#average_rate" do
     describe "when submission is unrated" do
-      let!(:submission) { FactoryGirl.build(:submission) }
+      let(:rates) { double }
+      subject { submission_presenter.average_rate }
 
-      it { expect(subject.average_rate).to eq(nil) }
-    end
+      before { allow(submission).to receive(:rated?).and_return(false) }
 
-    describe "when submission is rated and has no rates" do
-      before do
-        allow(Setting).to receive(:get).and_return(
-          FactoryGirl.create(:setting, required_rates_num: 0))
-      end
+      it { is_expected.to eq(nil) }
+    end
 
-      let!(:submission) do
-        FactoryGirl.build(:submission, :rated, required_rates_num: 0)
-      end
+    describe "when submission is rated" do
+      # overriden with real object since average is calculated in the db
+      let(:submission) { FactoryGirl.create(:submission, :rated, rate_value: 2) }
+      let(:rates) { submission.rates }
```

Comment:
@lesniakania I tried a couple of things but it seems to me that's the best choice. I changed the `average` to be calculated in the db, so there need to be db records, that's why I'm building these rates with FactoryGirls